src/Moq/Mock.cs

```diff
@@ -403,15 +405,18 @@ internal bool HasAnyInnerMocks()
 
 		internal IEnumerable<Mock> GetInnerMocks()
 		{
-			return this.InnerMocks.Values.Select(inner => inner.Mock);
+			foreach (var returnValue in this.InnerMocks.Values)
+			{
+				Mock.TryGetFromReturnValue(returnValue, out var innerMock);
```

Comment:
This is essentially the "unwrap" operation that `MockWithWrappedMockObject` previously prevented by caching the unwrapped mock next to the wrapped return value. I should check how performance is affected by having this caching removed.